### ISSUE TITLE
Yes but *what does it do*?

### ISSUE THREAD

**Issue by sdhull** [lifeart/ember-cli-fallback-builds#2]
From the Readme:
> check index.html
done!

This doesn't mean anything to me. What should I expect to see in `index.html`? What does this do? How does it work? I don't get it

**Comment by lifeart** [comment]
@sdhull 


`dist/index.html` before:

```html
<body>
    <script src="/assets/vendor.js"></script>
    <script src="/assets/dummy.js"></script>
</body>
```

after:

```html
<body>
    <script src="/assets/vendor.js" type="module"></script>
    <script src="/assets/dummy.js" type="module"></script>

    <script src="/assets/vendor.fb.js" nomodule></script>
    <script src="/assets/dummy.fb.js" nomodule></script>
</body>
```

new files: `vendor.fb.js`, `dummy.fb.js` - legacy js bundles (for older browsers)

**Comment by neojp** [comment]
Please correct me if I'm wrong, but from what I can understand, it creates a new set of `ember build` files with a different target from browserlist.

If you ever need to create a build to target certain browser (like IE) you can without bloating the main build with polyfills that probably won't ever be used.

This isn't only useful for IE, a lot new JS features like const/let and classes and arrow functions are available in the current version of browsers. You could use it to build a set of Ember with those features for current version of browsers which should have less KB due to less transpiling, then build a different set with regular transpiling to support the other browsers. You can always save a few KB here and there by using this technique, in the end everything adds up. It's not that simple, but if you reaaaally care about bundle size, now you can.

**Comment by lifeart** [comment]
@neojp yeah, you right. More how it's working: https://github.com/lifeart/ember-cli-fallback-builds/issues/3#issuecomment-509844029

**Comment by sdhull** [comment]
Thanks for the explanations! I'd love to see the Readme updated, I think a lot of folks will find this useful. Do you know how this approach behaves with fastboot?

**Comment by lifeart** [comment]
@sdhull sorry, no information about fastboot(.

**Comment by sdhull** [comment]
OK thanks... We'll find out the hard way when we get around to adding this to our build 😉

**Comment by sdhull** [comment]
Also thank you for this, TIL about [nomodule](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#Module_fallback)

**Comment by sdhull** [comment]
One more question: for browsers that don't support `ES2015` modules (eg, nomodule fallback), do they download _both_ js files, with the 2nd one essentially clobbering the 1st?

**Comment by lifeart** [comment]
@sdhull it depends on browser, https://github.com/philipwalton/webpack-esnext-boilerplate/issues/1
